src/Streamly/Internal/LZ4.hs

```diff
@@ -247,7 +247,7 @@ compressChunk cfg speed ctx arr = do
                 $ error $ "compressChunk: compressed length <= 0."
                     ++ " maxCompLenC: " ++ show maxCompLenC
                     ++ " uncompLenC: " ++ show uncompLenC
-              (MArray.Array fptr dstBegin dstMax) <-
+              (MArray.Array cont dstBegin_ dstBegin dstMax) <-
                   MArray.newArray (maxCompLen + metaSize_)
```

Comment:
Suggestion for future improvement: We should avoid using the array constructor. May need to add some more array APIs to avoid it.

Reply:
Added an issue here.